packages/smooth_app/lib/pages/product/common/product_list_page.dart

```diff
@@ -198,99 +200,32 @@ class _ProductListPageState extends State<ProductListPage>
-                                    onTap: _selectionMode ? onTap : null,
-                                    onLongPress: !_selectionMode
-                                        ? () => setState(
-                                            () => _selectionMode = true)
-                                        : null,
-                                  ),
-                                ),
-                              ],
-                            ),
-                          ),
-                        );
-                        if (dismissible) {
-                          return Dismissible(
-                            direction: DismissDirection.endToStart,
-                            background: Container(
-                              alignment: Alignment.centerRight,
-                              margin: const EdgeInsets.symmetric(vertical: 14),
-                              color: RED_COLOR,
-                              padding: const EdgeInsets.only(right: 30),
-                              child: const Icon(
-                                Icons.delete,
-                                color: Colors.white,
-                              ),
-                            ),
-                            key: Key(product.barcode!),
-                            onDismissed:
-                                (final DismissDirection direction) async {
-                              final bool removed =
-                                  productList.remove(product.barcode!);
-                              if (removed) {
-                                daoProductList.put(productList);
-                                _selectedBarcodes.remove(product.barcode);
-                                setState(() => products.removeAt(index));
-                              }
-                              if (!mounted) {
-                                return;
-                              }
-                              ScaffoldMessenger.of(context).showSnackBar(
-                                SnackBar(
-                                  content: Text(
-                                    removed
-                                        ? appLocalizations
-                                            .product_removed_history
-                                        : appLocalizations
-                                            .product_could_not_remove,
-                                  ),
-                                  duration: const Duration(seconds: 3),
-                                ),
-                              );
-                              // TODO(monsieurtanuki): add a snackbar ("put back the food")
-                            },
-                            child: child,
+                    child: Consumer<UpToDateProductProvider>(
+                      builder: (
+                        _,
+                        final UpToDateProductProvider provider,
+                        __,
```

Comment:
@M123-dev Yeah I know, the idea behind this kind of syntax in `dart` is to avoid to add a pointless and confusing nth version of `BuildContext`. Not a big fan of that syntax either.